src/main/java/org/owasp/esapi/codecs/AbstractCodec.java

```diff
@@ -58,8 +58,10 @@ public AbstractCodec() {
     /**
      * {@inheritDoc}
      * </p><p>
-     * <b>WARNING!!</b>  {@code Character} based {@code Codec}s will silently transform code points that are not
-     * legal UTF code points into garbage data as they will cast them to {@code char}s.
+     * <b>WARNING!!</b>  {@code Character} based {@code Codec}s will only handle the byte range of
+     * 0-65535 (0x0-0xffff).  Passing any data represented by a higher numerical value will result in
```

Comment:
Oy.  So much for a quiet morning!

Okay, so I made a mistake going all the way back to 2017.  Because of the issue I was resolving involving non-BMP characters,  I created the AbstractIntegerCodec to be able to handle the complete UTF-8 range (and future proof it to boot.)  

My underlying assumption all this time was that because of autoboxing, that the `Character` class was 1:1 with `char`.  `char`'s range is 0x00-0xffff.  https://docs.oracle.com/javase/tutorial/java///nutsandbolts/datatypes.html

*this* library before I created the abstract classes was using `char` and `char` alone without using the character classes.  So my framing of the problem began here and is situated with not having read the docs on the Character class, believing it to be ultimately 1:1 with the `char` datatype.  

THAT all being said, I was fully in possession of the fact that under the hood, characters in Java are UTF-16 encoded bytes, and that hidden layer never disappears (and makes you constantly second-guess your code.)  

The end result of all of this is that we support the full range of that java `Character` class documentation.  We haven't necessarily tested the entire character set and I've had a sticky note to do that since 2017.  

The documentation should be updated to reflect what's in the `Character` class.   I still have full faith in the AbstractInteger implementation but the AbstractCharacter class was never tested beyond ensuring that regression tests passed.  Which absolutely means that nothing beyond 0xffff was ever tested.  (Supported, but not tested, and I'm sorry for the error.)

Reply:
@xeno6696 - Okay then. That's not as bad as I thought. Seems like everything should work. However, if you haven't done so already, could you create a GitHub issue to outline specifically what needs tested. It seems conceptually simple enough that we would even label it as "Good First Issue", but I don't want it to fall through the cracks for another 5 years or so.

In the meantime, I'm going to merge this PR.